### ISSUE TITLE
Lite version of the project

### ISSUE THREAD

**Issue by mehdi-najaran** [YahnisElsts/plugin-update-checker#553]
Hello, is it possible for you to provide me with a light and compact version of the project?

A version without the ability to connect to GitHub
Version without comments
Special version for updating the plugin (without theme)

In this way, less code and volume is occupied in the project

**Comment by mehdi-najaran** [comment]
Now I put the project file in my plugin core

This issue has increased its volume, am I wrong?

**Comment by YahnisElsts** [comment]
You are not wrong. By adding PUC to your plugin, the size of your plugin download has increased by about 150 KB (zipped). 

But what is the meaningful impact of that? Your users will have to spend a fraction of a second longer downloading your plugin, and it will take up about 360 KB more space on their server. Will they even notice? How much does that matter, and how much time are you willing to spend to optimize it? Personally, it doesn't look very significant to me.

(I'm aware that this sounds suspiciously similar to the kind of reasoning that some careless people might use to justify e.g. websites adding megabytes of JS to every page, but I think scale does matter here. Installing a plugin is generally a one-time kind of thing, or close to that.)

**Comment by YahnisElsts** [comment]
All right.

**Comment by DavidAnderson684** [comment]
Removing parts of a project can be unexpectedly dangerous: if multiple plugins are using this library, and the one that gets loaded is a version with some classes removed, then that may cause the plugins that expected the full version to be present to have a problem.

**Comment by YahnisElsts** [comment]
That is also a good point, and one that didn't occur to me. Though to be fair, it would be possible to redesign the library to still work correctly in that situation - it would just take even more work.

**Comment by theo-gk** [comment]
@mehdi-najaran You can safely remove the translation files from the /languages/ directory in case you don't need them. It will save you more than 80KB. But it will have no effect in performance, just in file size. Probably isn't worth it, though.
